### ISSUE TITLE
Running tests failed

### ISSUE THREAD

**Issue by lonsty** [karec/cookiecutter-flask-restful#21]
#### I created a project using coockiecutter from this template, then:
```
cd myproject
virtualenv --no-site-packages --pyhton=python3.6 venv
source venv/bin/activate
pip install -r requirements.txt
pip install -e .

myapi init
```
#### But when i running tests 
```
tox
```

#### There were a lot of errors appeared, below is the test log:
```
tox
GLOB sdist-make: /mnt/f/workspace/git/FlaskProjectTemplate/ig-restful-api/setup.py
python create: /mnt/f/workspace/git/FlaskProjectTemplate/ig-restful-api/.tox/python
python installdeps: flake8, pytest, pytest-flask, pytest-runner, pytest-factoryboy, factory_boy, -rrequirements.txt
python inst: /mnt/f/workspace/git/FlaskProjectTemplate/ig-restful-api/.tox/.tmp/package/1/igapi-0.1.zip
python installed: alembic==1.0.11,aniso8601==7.0.0,atomicwrites==1.3.0,attrs==19.1.0,Click==7.0,entrypoints==0.3,factory-boy==2.12.0,Faker==2.0.0,filelock==3.0.12,flake8==3.7.8,Flask==1.1.1,Flask-JWT-Extended==3.21.0,flask-marshmallow==0.10.1,Flask-Migrate==2.5.2,Flask-RESTful==0.3.7,Flask-SQLAlchemy==2.4.0,igapi==0.1,importlib-metadata==0.19,inflection==0.3.1,itsdangerous==1.1.0,Jinja2==2.10.1,Mako==1.1.0,MarkupSafe==1.1.1,marshmallow==3.0.0,marshmallow-sqlalchemy==0.17.0,mccabe==0.6.1,more-itertools==7.2.0,packaging==19.1,passlib==1.7.1,pathlib2==2.3.4,pluggy==0.12.0,py==1.8.0,pycodestyle==2.5.0,pyflakes==2.1.1,PyJWT==1.7.1,pyparsing==2.4.2,pytest==5.1.0,pytest-factoryboy==2.0.3,pytest-flask==0.15.0,pytest-runner==5.1,python-dateutil==2.8.0,python-dotenv==0.10.3,python-editor==1.0.4,pytz==2019.2,six==1.12.0,SQLAlchemy==1.3.7,text-unidecode==1.2,toml==0.10.0,tox==3.13.2,virtualenv==16.7.3,wcwidth==0.1.7,Werkzeug==0.15.5,zipp==0.5.2
python run-test-pre: PYTHONHASHSEED='4188641553'
python run-test: commands[0] | flake8 igapi
python run-test: commands[1] | pytest tests
=============================================================================== test session starts ================================================================================
platform linux -- Python 3.5.2, pytest-5.1.0, py-1.8.0, pluggy-0.12.0
cachedir: .tox/python/.pytest_cache
rootdir: /mnt/f/workspace/git/FlaskProjectTemplate/ig-restful-api
plugins: factoryboy-2.0.3, flask-0.15.0
collected 7 items

tests/test_auth.py ..                                                                                                                                                        [ 28%]
tests/test_user.py FF.FF                                                                                                                                                     [100%]

===================================================================================== FAILURES =====================================================================================
__________________________________________________________________________________ test_get_user ___________________________________________________________________________________

client = <FlaskClient <Flask 'igapi'>>, db = <SQLAlchemy engine=sqlite:///:memory:>, user = <User user1>
admin_headers = {'authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZGVudGl0eSI6MSwiZXhwIjoxNTY2MTg0OTI0LCJpYXQiOjE1NjY...oxNTY2MTg0MDI0LCJ0eXBlIjoiYWNjZXNzIn0.qYzgV4glilctyzHvsBJSb6OnicTPJRskbOb3hJhRMds', 'content-type': 'application/json'}

>   ???

tests/test_user.py:283:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1029: in get
    return self.open(*args, **kw)
.tox/python/lib/python3.5/site-packages/flask/testing.py:227: in open
    follow_redirects=follow_redirects,
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:993: in open
    response = self.run_wsgi_app(environ.copy(), buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:884: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1119: in run_wsgi_app
    app_rv = app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2463: in __call__
    return self.wsgi_app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2449: in wsgi_app
    response = self.handle_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1866: in handle_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:2446: in wsgi_app
    response = self.full_dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1951: in full_dispatch_request
    rv = self.handle_user_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1820: in handle_user_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:1949: in full_dispatch_request
    rv = self.dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1935: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:458: in wrapper
    resp = resource(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask/views.py:89: in view
    return self.dispatch_request(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:573: in dispatch_request
    resp = meth(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_jwt_extended/view_decorators.py:103: in wrapper
    return fn(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <igapi.api.resources.user.UserResource object at 0x7f87a3d18438>, user_id = 2

    def get(self, user_id):
        schema = UserSchema()
        user = User.query.get_or_404(user_id)
>       return {"user": schema.dump(user).data}
E       AttributeError: 'dict' object has no attribute 'data'

igapi/api/resources/user.py:27: AttributeError
__________________________________________________________________________________ test_put_user ___________________________________________________________________________________

client = <FlaskClient <Flask 'igapi'>>, db = <SQLAlchemy engine=sqlite:///:memory:>, user = <User updated>
admin_headers = {'authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZGVudGl0eSI6MSwiZXhwIjoxNTY2MTg0OTI0LCJpYXQiOjE1NjY...oxNTY2MTg0MDI0LCJ0eXBlIjoiYWNjZXNzIn0.mjjdw9SzzuvAvouJyuQi96GoQurv4WWKdOy8GMU93Uo', 'content-type': 'application/json'}

    def test_put_user(client, db, user, admin_headers):
        # test 404
        rep = client.put("/api/v1/users/100000", headers=admin_headers)
        assert rep.status_code == 404

        db.session.add(user)
        db.session.commit()

        data = {'username': 'updated'}

        # test update user
        rep = client.put(
            '/api/v1/users/%d' % user.id,
            json=data,
>           headers=admin_headers
        )

tests/test_user.py:50:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1049: in put
    return self.open(*args, **kw)
.tox/python/lib/python3.5/site-packages/flask/testing.py:227: in open
    follow_redirects=follow_redirects,
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:993: in open
    response = self.run_wsgi_app(environ.copy(), buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:884: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1119: in run_wsgi_app
    app_rv = app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2463: in __call__
    return self.wsgi_app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2449: in wsgi_app
    response = self.handle_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1866: in handle_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:2446: in wsgi_app
    response = self.full_dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1951: in full_dispatch_request
    rv = self.handle_user_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1820: in handle_user_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:1949: in full_dispatch_request
    rv = self.dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1935: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:458: in wrapper
    resp = resource(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask/views.py:89: in view
    return self.dispatch_request(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:573: in dispatch_request
    resp = meth(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_jwt_extended/view_decorators.py:103: in wrapper
    return fn(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <igapi.api.resources.user.UserResource object at 0x7f87a3c7da20>, user_id = 2

    def put(self, user_id):
        schema = UserSchema(partial=True)
        user = User.query.get_or_404(user_id)
>       user, errors = schema.load(request.json, instance=user)
E       TypeError: 'User' object is not iterable

igapi/api/resources/user.py:32: TypeError
_________________________________________________________________________________ test_create_user _________________________________________________________________________________

client = <FlaskClient <Flask 'igapi'>>, db = <SQLAlchemy engine=sqlite:///:memory:>
admin_headers = {'authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZGVudGl0eSI6MSwiZXhwIjoxNTY2MTg0OTI0LCJpYXQiOjE1NjY...oxNTY2MTg0MDI0LCJ0eXBlIjoiYWNjZXNzIn0.0QCtn7uKt6eTjby9DA6ZvdjFn1Sach2uA-s5z8lUoy0', 'content-type': 'application/json'}

    def test_create_user(client, db, admin_headers):
        # test bad data
        data = {
            'username': 'created'
        }
        rep = client.post(
            '/api/v1/users',
            json=data,
>           headers=admin_headers
        )

tests/test_user.py:86:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1039: in post
    return self.open(*args, **kw)
.tox/python/lib/python3.5/site-packages/flask/testing.py:227: in open
    follow_redirects=follow_redirects,
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:993: in open
    response = self.run_wsgi_app(environ.copy(), buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:884: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1119: in run_wsgi_app
    app_rv = app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2463: in __call__
    return self.wsgi_app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2449: in wsgi_app
    response = self.handle_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1866: in handle_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:2446: in wsgi_app
    response = self.full_dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1951: in full_dispatch_request
    rv = self.handle_user_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1820: in handle_user_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:1949: in full_dispatch_request
    rv = self.dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1935: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:458: in wrapper
    resp = resource(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask/views.py:89: in view
    return self.dispatch_request(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:573: in dispatch_request
    resp = meth(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_jwt_extended/view_decorators.py:103: in wrapper
    return fn(*args, **kwargs)
igapi/api/resources/user.py:60: in post
    user, errors = schema.load(request.json)
.tox/python/lib/python3.5/site-packages/marshmallow_sqlalchemy/schema.py:216: in load
    return super(ModelSchema, self).load(data, *args, **kwargs)
.tox/python/lib/python3.5/site-packages/marshmallow/schema.py:684: in load
    data, many=many, partial=partial, unknown=unknown, postprocess=True
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <UserSchema(many=False)>, data = {'username': 'created'}

    def _do_load(
        self, data, *, many=None, partial=None, unknown=None, postprocess=True
    ):
        """Deserialize `data`, returning the deserialized result.

        :param data: The data to deserialize.
        :param bool many: Whether to deserialize `data` as a collection. If `None`, the
            value for `self.many` is used.
        :param bool|tuple partial: Whether to validate required fields. If its
            value is an iterable, only fields listed in that iterable will be
            ignored will be allowed missing. If `True`, all fields will be allowed missing.
            If `None`, the value for `self.partial` is used.
        :param unknown: Whether to exclude, include, or raise an error for unknown
            fields in the data. Use `EXCLUDE`, `INCLUDE` or `RAISE`.
            If `None`, the value for `self.unknown` is used.
        :param bool postprocess: Whether to run post_load methods..
        :return: A dict of deserialized data
        :rtype: dict
        """
        error_store = ErrorStore()
        errors = {}
        many = self.many if many is None else bool(many)
        unknown = unknown or self.unknown
        if partial is None:
            partial = self.partial
        # Run preprocessors
        if self._has_processors(PRE_LOAD):
            try:
                processed_data = self._invoke_load_processors(
                    PRE_LOAD, data, many=many, original_data=data, partial=partial
                )
            except ValidationError as err:
                errors = err.normalized_messages()
                result = None
        else:
            processed_data = data
        if not errors:
            # Deserialize data
            result = self._deserialize(
                processed_data,
                error_store=error_store,
                many=many,
                partial=partial,
                unknown=unknown,
            )
            # Run field-level validation
            self._invoke_field_validators(
                error_store=error_store, data=result, many=many
            )
            # Run schema-level validation
            if self._has_processors(VALIDATES_SCHEMA):
                field_errors = bool(error_store.errors)
                self._invoke_schema_validators(
                    error_store=error_store,
                    pass_many=True,
                    data=result,
                    original_data=data,
                    many=many,
                    partial=partial,
                    field_errors=field_errors,
                )
                self._invoke_schema_validators(
                    error_store=error_store,
                    pass_many=False,
                    data=result,
                    original_data=data,
                    many=many,
                    partial=partial,
                    field_errors=field_errors,
                )
            errors = error_store.errors
            # Run post processors
            if not errors and postprocess and self._has_processors(POST_LOAD):
                try:
                    result = self._invoke_load_processors(
                        POST_LOAD,
                        result,
                        many=many,
                        original_data=data,
                        partial=partial,
                    )
                except ValidationError as err:
                    errors = err.normalized_messages()
        if errors:
            exc = ValidationError(errors, data=data, valid_data=result)
            self.handle_error(exc, data, many=many, partial=partial)
>           raise exc
E           marshmallow.exceptions.ValidationError: {'password': ['Missing data for required field.'], 'email': ['Missing data for required field.']}

.tox/python/lib/python3.5/site-packages/marshmallow/schema.py:842: ValidationError
________________________________________________________________________________ test_get_all_user _________________________________________________________________________________

client = <FlaskClient <Flask 'igapi'>>, db = <SQLAlchemy engine=sqlite:///:memory:>, user_factory = <class 'tests.test_user.UserFactory'>
admin_headers = {'authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZGVudGl0eSI6MSwiZXhwIjoxNTY2MTg0OTI0LCJpYXQiOjE1NjY...oxNTY2MTg0MDI0LCJ0eXBlIjoiYWNjZXNzIn0.F9D-sbfjg_jSmWcW0kkIfHBBgyfqHFXk9K1t347wWZs', 'content-type': 'application/json'}

    def test_get_all_user(client, db, user_factory, admin_headers):
        users = user_factory.create_batch(30)

        db.session.add_all(users)
        db.session.commit()

>       rep = client.get('/api/v1/users', headers=admin_headers)

tests/test_user.py:113:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1029: in get
    return self.open(*args, **kw)
.tox/python/lib/python3.5/site-packages/flask/testing.py:227: in open
    follow_redirects=follow_redirects,
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:993: in open
    response = self.run_wsgi_app(environ.copy(), buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:884: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
.tox/python/lib/python3.5/site-packages/werkzeug/test.py:1119: in run_wsgi_app
    app_rv = app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2463: in __call__
    return self.wsgi_app(environ, start_response)
.tox/python/lib/python3.5/site-packages/flask/app.py:2449: in wsgi_app
    response = self.handle_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1866: in handle_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:2446: in wsgi_app
    response = self.full_dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1951: in full_dispatch_request
    rv = self.handle_user_exception(e)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:269: in error_router
    return original_handler(e)
.tox/python/lib/python3.5/site-packages/flask/app.py:1820: in handle_user_exception
    reraise(exc_type, exc_value, tb)
.tox/python/lib/python3.5/site-packages/flask/_compat.py:38: in reraise
    raise value.with_traceback(tb)
.tox/python/lib/python3.5/site-packages/flask/app.py:1949: in full_dispatch_request
    rv = self.dispatch_request()
.tox/python/lib/python3.5/site-packages/flask/app.py:1935: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:458: in wrapper
    resp = resource(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask/views.py:89: in view
    return self.dispatch_request(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_restful/__init__.py:573: in dispatch_request
    resp = meth(*args, **kwargs)
.tox/python/lib/python3.5/site-packages/flask_jwt_extended/view_decorators.py:103: in wrapper
    return fn(*args, **kwargs)
igapi/api/resources/user.py:56: in get
    return paginate(query, schema)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

query = <flask_sqlalchemy.BaseQuery object at 0x7f87a3addba8>, schema = <UserSchema(many=True)>

    def paginate(query, schema):
        page = request.args.get('page', DEFAULT_PAGE_NUMBER)
        per_page = request.args.get('page_size', DEFAULT_PAGE_SIZE)
        page_obj = query.paginate(page=page, per_page=per_page)
        next = url_for(
            request.endpoint,
            page=page_obj.next_num if page_obj.has_next else page_obj.page,
            per_page=per_page,
            **request.view_args
        )
        prev = url_for(
            request.endpoint,
            page=page_obj.prev_num if page_obj.has_prev else page_obj.page,
            per_page=per_page,
            **request.view_args
        )

        return {
            'total': page_obj.total,
            'pages': page_obj.pages,
            'next': next,
            'prev': prev,
>           'results': schema.dump(page_obj.items).data
        }
E       AttributeError: 'list' object has no attribute 'data'

igapi/commons/pagination.py:31: AttributeError
=========================================================================== 4 failed, 3 passed in 2.88s ============================================================================
ERROR: InvocationError for command /mnt/f/workspace/git/FlaskProjectTemplate/ig-restful-api/.tox/python/bin/pytest tests (exited with code 1)
_____________________________________________________________________________________ summary ______________________________________________________________________________________
ERROR:   python: commands failed
```

#### The following packages is currently in use:
```
alembic==1.0.11
aniso8601==7.0.0
attrs==19.1.0
Click==7.0
filelock==3.0.12
Flask==1.1.1
Flask-JWT-Extended==3.21.0
flask-marshmallow==0.10.1
Flask-Migrate==2.5.2
Flask-RESTful==0.3.7
Flask-SQLAlchemy==2.4.0
# Editable install with no version control (igapi==0.1)
-e /mnt/f/workspace/git/FlaskProjectTemplate/ig-restful-api
importlib-metadata==0.19
itsdangerous==1.1.0
Jinja2==2.10.1
Mako==1.1.0
MarkupSafe==1.1.1
marshmallow==3.0.0
marshmallow-sqlalchemy==0.17.0
packaging==19.1
passlib==1.7.1
pluggy==0.12.0
py==1.8.0
PyJWT==1.7.1
pyparsing==2.4.2
python-dateutil==2.8.0
python-dotenv==0.10.3
python-editor==1.0.4
pytz==2019.2
six==1.12.0
SQLAlchemy==1.3.7
toml==0.10.0
tox==3.13.2
virtualenv==16.7.3
Werkzeug==0.15.
```
### May i ask which packages should work correctly?

**Comment by karec** [comment]
This is due to the new marshmallow release (3) that have been released. Simplest way to fix that bug is to pin marshmallow to `<3` until I merge updates for this new version.

**Comment by karec** [comment]
Should be fixed with #20 (with swagger support too), I'm going to work on #4 to include marshmallow 3 support

**Comment by lonsty** [comment]
Thank you, the problem is solved.
